opensearch-api/lib/opensearch/api/actions/security/change_password.rb

```diff
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+#
+# Licensed to OpenSearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. OpenSearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+module OpenSearch
+  module API
+    module Security
+      module Actions
+        # Changes the passwords of users in the native realm and built-in users.
+        #
+        # @option arguments [String] :username The username of the user to change the password for
+        # @option arguments [String] :refresh If `true` (the default) then refresh the affected shards to make this operation visible to search, if `wait_for` then wait for a refresh to make this operation visible to search, if `false` then do nothing with refreshes. (options: true, false, wait_for)
+        # @option arguments [Hash] :headers Custom HTTP headers
+        # @option arguments [String] :password the new password for the user (*Required*)
+        #
+        #
+        def change_password(arguments = {})
+          patch_user(arguments.merge({
+            username: arguments.delete(:username),
```

Comment:
Use `.fetch`. However here we merge `username` from `arguments` back to `arguments` which is a noop.

Can you please check whether we do `delete` all over the place? We should not be modifying hash data coming in. Open an issue if we do and don't have one open. See https://code.dblock.org/2020/11/11/dup-dont-modify-input-hash-parameters-in-ruby.html for an example of how this is a bad idea.